components/accept-financial-contributions/HostCollectiveCard.js

```diff
@@ -0,0 +1,259 @@
+import React, { useState, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { Flex } from '@rebass/grid';
+import { FormattedDate, useIntl, defineMessages, FormattedMessage } from 'react-intl';
+import { graphql } from '@apollo/react-hoc';
+import { get } from 'lodash';
+
+import { API_V2_CONTEXT, gqlV2 } from '../../lib/graphql/helpers';
+import { getErrorFromGraphqlException } from '../../lib/errors';
+
+import Container from '../Container';
+import { P, H1 } from '../Text';
+import StyledButton from '../StyledButton';
+import Currency from '../Currency';
+import StyledCollectiveCard from '../StyledCollectiveCard';
+import Modal, { ModalBody, ModalHeader, ModalFooter } from '../StyledModal';
+import Avatar from '../Avatar';
+import HTMLContent from '../HTMLContent';
+import MessageBox from '../MessageBox';
+import StyledCheckbox from '../StyledCheckbox';
+import ExternalLink from '../ExternalLink';
+
+const messages = defineMessages({
+  collectives: {
+    id: 'pricingTable.row.collectives',
+    defaultMessage: 'Collectives',
+  },
+  contributors: {
+    id: 'ContributorsFilter.Financial',
+    defaultMessage: 'Financial contributors',
+  },
+  budget: {
+    id: 'YearlyBudget',
+    defaultMessage: 'Yearly budget',
+  },
+  apply: {
+    id: 'host.apply.create.btn',
+    defaultMessage: 'Apply',
+  },
+  currency: {
+    id: 'collective.currency.label',
+    defaultMessage: 'Currency',
+  },
+  hostFee: {
+    id: 'HostFee',
+    defaultMessage: 'Host fee',
+  },
+  hostSince: {
+    id: 'HostSince',
+    defaultMessage: 'Host since',
+  },
+  cancel: {
+    id: 'actions.cancel',
+    defaultMessage: 'Cancel',
+  },
+  submit: {
+    id: 'actions.submitApplication',
+    defaultMessage: 'Submit application',
+  },
+  tosError: {
+    id: 'acceptContributions.tosError',
+    defaultMessage: "Please accept the host's terms of service.",
+  },
+});
+
+const HostCollectiveCard = ({ hostCollective, applyingCollective, ...props }) => {
```

Comment:
For this component I'd recommend using the `useMutation` hook rather than `graphql`. It will provide you with a `loading` flag and an `error` out of the box.

Reply:
Will add tomorrow